arbnode/transaction_streamer.go

```diff
@@ -1775,25 +1736,53 @@ func (s *TransactionStreamer) espressoSwitch(ctx context.Context, ignored struct
 		return retryRate
 	}
 	espressoMerkleProofEphemeralErrorHandler.Reset()
+	return s.espressoTxnsPollingInterval
```

Comment:
I think we should maybe return 0 in case a transaction was finalized and return the polling interval if it wasn't. Because if one transaction was finalized there might be more that are already ready. Only if we check and nothing new has been finalized do we need to wait some time